### DISCUSSION TITLE
Artificial Gravity like Starminer Mod

### DISCUSSION THREAD

**Discussion by SebaSphere** [TeamGalacticraft/Galacticraft#173] [Feature Requests]
If it would be possible to implement the Starminer mod (https://m.youtube.com/watch?v=mkKjlW3EnoA), that could be cool. Instead of the gravity being implemented everywhere, setting it so only you can make artificial gravity spheres in space stations and zero g environments could be cool.

Adjustable gravity within the sphere could be practical.

**Comment by vos6434** [comment]
need more details, Starminers gravity sucked imo.

**Comment by craftablescience** [comment]
Galacticraft focuses more on different dimensions, not having planets hanging over the Overworld. I can see this being useful on a spinning Space Station though.

**Comment by SebaSphere** [comment]
@vos6434 edited description to be more detailed. Was in the middle of writing when you said that lol.

**Comment by SebaSphere** [comment]
@craftablescience was in the middle of writing space stations only accually lmao.

**Comment by craftablescience** [comment]
I think it would be more useful to implement spinning stations, with centripetal force and inertia pulling on you instead of near-future technology.

**Comment by SebaSphere** [comment]
@craftablescience While that is a great idea, I feel like it might be harder to implement so that the player accually goes up to the space station dimension and place it themselves. I feel like a pregenerated spinning station would kind of take away from building your own things but If we could find a way to implement a spinning station in a non buggy and functional way, that would be cool. I do know that @micdoodle8 (May be radfast I forgot), did make a functioning station rotating around Jupiter. Honestly a cool concept but how could that be implemented to where you have full control how it’s crafted. A concept I think might be cool is that if you wanna adjust gravity, you could add and remove heavy blocks to keep it “more realistic”

**Comment by craftablescience** [comment]
Remember the spinning rockets in GC 4.0? They already spin space stations. They just didn't generate gravity

**Comment by craftablescience** [comment]
And it wouldn't be pregenerated

**Comment by craftablescience** [comment]
But I do see the issue with circles in Minecraft, that would be difficult

**Comment by SebaSphere** [comment]
@craftablescience I think that it just rotates the sky box of the whole dimension lol. I could see that the thrusters may need to be reworked completely tbh. Circles would really limit who has acsess to artificial gravity and things inside the space station won’t look good at all under that system. The testing build of the spinning station was really cool but it’s pregenerated.

**Comment by craftablescience** [comment]
Yep I know, you make good points but it all comes down to the fact that gravity is fundamentally circular, and no system  using blocks will be able to accurately simulate it

**Comment by SebaSphere** [comment]
Yep that is completely true. I do remember Starminer having a mode where it was essentially a square planet. That could work to solve it @craftablescience . 

I think a rotating space station would work if the gravity is completely redone. It would be really cool to have it so you press “q and e” or (something like that idk) in order to flip your position. If you are able to do that, it would probaly make a spinning space station more easy to navigate. Starminer would probaly be more easier to implement but a spinning space station I’d imagine would be harder to dev.

To help with building, some sort of open blocks like enchanced builder could work

**Comment by NightScale5755** [comment]
hey guys. I know I'm pretty late to this.. But a more practical approach would be to have a block, that only works in sealed rooms, much like the oxygen sealer. Within these rooms with the machine, overworld gravity would be implemented.
I hope this idea of mine is helpful.
